runtime/README.rst

```diff
@@ -3,35 +3,95 @@
-- `extensions <https://github.com/PennyLaneAI/catalyst/tree/main/runtime/extensions>`_ : A collection of extensions for backend simulators to fit into the `QIR programming model <https://github.com/qir-alliance/qir-spec/blob/main/specification/v0.1/4_Quantum_Runtime.md#qubits>`_. The `StateVectorDynamicCPU <https://github.com/PennyLaneAI/catalyst/tree/main/runtime/extensions/StateVectorDynamicCPU.hpp>`_ class extends the state-vector class of `Pennylane-Lightning <https://github.com/PennyLaneAI/pennylane-lightning>`_ providing dynamic allocation and deallocation of qubits.
-- `lib <https://github.com/PennyLaneAI/catalyst/tree/main/runtime/lib>`_ : The core modules of the runtime are structured into ``lib/capi`` and ``lib/backend``. `lib/capi <https://github.com/PennyLaneAI/catalyst/tree/main/runtime/lib/capi>`_  contains the source of the bridge between QIR instructions in LLVM-IR and C++ backends. `lib/backend <https://github.com/PennyLaneAI/catalyst/tree/main/runtime/lib/backend>`_ contains implementations of the ``QuantumDevice`` API for backend simulators.
-- `tests <https://github.com/PennyLaneAI/catalyst/tree/main/runtime/tests>`_ : A collection of C++ tests for modules and methods in the runtime.
+- `include <https://github.com/PennyLaneAI/catalyst/tree/main/runtime/include>`_:
+    This contains the public header files of the runtime including the ``QuantumDevice`` API
+    for backend quantum devices and the runtime CAPI.
+
+- `extensions <https://github.com/PennyLaneAI/catalyst/tree/main/runtime/extensions>`_:
+    A collection of extensions for backend simulators to fit into the
+    `QIR programming model <https://github.com/qir-alliance/qir-spec/blob/main/specification/v0.1/4_Quantum_Runtime.md#qubits>`_.
+    The `StateVectorDynamicCPU <https://github.com/PennyLaneAI/catalyst/tree/main/runtime/extensions/StateVectorDynamicCPU.hpp>`_
+    class extends the state-vector class of `Pennylane-Lightning <https://github.com/PennyLaneAI/pennylane-lightning>`_ providing
+    dynamic allocation and deallocation of qubits.
+
+- `lib <https://github.com/PennyLaneAI/catalyst/tree/main/runtime/lib>`_:
+    The core modules of the runtime are structured into ``lib/capi`` and ``lib/backend``.
+    `lib/capi <https://github.com/PennyLaneAI/catalyst/tree/main/runtime/lib/capi>`_  contains the source of the bridge between
+    QIR instructions in LLVM-IR and C++ backends. `lib/backend <https://github.com/PennyLaneAI/catalyst/tree/main/runtime/lib/backend>`_
+    contains implementations of the ``QuantumDevice`` API for backend simulators.
+
+- `tests <https://github.com/PennyLaneAI/catalyst/tree/main/runtime/tests>`_:
+    A collection of C++ tests for modules and methods in the runtime.
+
+Backend Devices
+===============
+
+A new device can be added to the list of supported ones in the runtime by implementing the quantum device API.
```

Comment:
As this section doesn't yet show how to integrate a new device into the runtime:
```suggestion
New device backends for the runtime can be realized by implementing the quantum device interface.
```